<!-- This file can be used to launch an playground world with a turtlebot in it -->
<launch>
	<arg name="3d_sensor" default="$(env TURTLEBOT_3D_SENSOR)"/>  <!-- r200, kinect, asus_xtion_pro -->

	<arg name="world_file" default="$(find turtlebot_gazebo)/worlds/playground.world"/>
	<arg name="map_file" default="$(find turtlebot_gazebo)/maps/playground.yaml"/>

	<arg name="robot_radius" default="0.3"/><!-- Originally found in costmap_common_params.yaml -->
	<arg name="xy_goal_tolerance" default="0.3"/><!-- Originally found in dwa_local_planner_params.yaml -->
	<arg name="oscillation_timeout" default="5.0"/><!-- Originally found in move_base_params.yaml -->
	<arg name="oscillation_distance" default="0.05"/><!-- Originally found in move_base_params.yaml -->

	<arg name="initial_x" default="0.0"/>
	<arg name="initial_y" default="0.0"/>
	<arg name="initial_a" default="0.0"/>

	<param name="initial_x" value="$(arg initial_x)" />
	<param name="initial_y" value="$(arg initial_y)" />
	<param name="initial_a" value="$(arg initial_a)" />

	<arg name="goal_x" default="-2.0"/>
	<arg name="goal_y" default="2.0"/>
	<arg name="goal_a" default="1.573"/>

	<param name="goal_x" value="$(arg goal_x)" />
	<param name="goal_y" value="$(arg goal_y)" />
	<param name="goal_a" value="$(arg goal_a)" />

	<!--  Launch turtle bot world -->
	<!-- <include file="$(find turtlebot_gazebo)/launch/turtlebot_world.launch"/> -->

	<!--  Launch turtle bot with custom world file -->
	<include file="$(find system1)/launch/includes/robot.launch.xml">
		<arg name="world_file" value="$(arg world_file)"/>
		<arg name="3d_sensor" value="$(arg 3d_sensor)"/>
		<arg name="initial_x" value="$(arg initial_x)"/>
		<arg name="initial_y" value="$(arg initial_y)"/>
		<arg name="initial_a" value="$(arg initial_a)"/>
	</include>

	<node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)"/>

	<node name="tf" pkg="tf" type="static_transform_publisher" args="$(arg initial_x) $(arg initial_y) $(arg initial_a) 0 0 0 /map /odom 100" />

	<include file="$(find turtlebot_rviz_launchers)/launch/view_robot.launch"/>

	<!-- Launch send goals node -->
	<node name="system1_node" pkg="system1" type="system1_node" output="screen"/>

</launch>